### PR TITLE
Adds proof harnesses for aws_byte_cursor_eq_byte_buf* functions

### PR DESCRIPTION
**Pull request by feliperodri**
*Description of changes:*
1. Updates implementation of `aws_byte_cursor_eq_byte_buf*` functions with pre- and post-conditions;
2. Adds proof harnesses for `aws_byte_cursor_eq_byte_buf*` functions;

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by feliperodri** [comment]
@tautschnig @karkhaz @angelhof @danielsn @markrtuttle could you review the proof harnesses?

File: .cbmc-batch/jobs/aws_byte_cursor_eq_byte_buf/aws_byte_cursor_eq_byte_buf_harness.c [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        assert(cur.len == buf.len);
+        if (cur.len > 0) {
+            assert_bytes_match(cur.ptr, buf.buffer, cur.len);
+        }
+    }
+
+    /* assertions */
+    assert(aws_byte_cursor_is_valid(&cur));
+    assert(aws_byte_buf_is_valid(&buf));
+    if (cur.len != 0) {
```
Comment (tautschnig) [inline .cbmc-batch/jobs/aws_byte_cursor_eq_byte_buf/aws_byte_cursor_eq_byte_buf_harness.c]: Nit pick: above you use `> 0` and here you use `!= 0` - I think those should be consistent.

File: source/byte_buf.c [hunk 2]
```diff
@@ -416,14 +416,20 @@ uint64_t aws_hash_byte_cursor_ptr_ignore_case(const void *item) {
...
+    AWS_POSTCONDITION(aws_byte_cursor_is_valid(a) && aws_byte_buf_is_valid(b));
+    return rv;
 }
 
-bool aws_byte_cursor_eq_byte_buf_ignore_case(const struct aws_byte_cursor *a, const struct aws_byte_buf *b) {
-    AWS_ASSERT(a && b);
-    return aws_array_eq_ignore_case(a->ptr, a->len, b->buffer, b->len);
+bool aws_byte_cursor_eq_byte_buf_ignore_case(
+    const struct aws_byte_cursor *const a,
+    const struct aws_byte_buf *const b) {
```
Comment (tautschnig) [inline source/byte_buf.c]: Formatting change?

Comment (feliperodri) [reply source/byte_buf.c]: The line size increased because of the new `*const` ones.

**Pull request merged**